agent/engine/engine_unix_integ_test.go

```diff
@@ -938,6 +938,8 @@ func TestStartStopWithSecurityOptionNoNewPrivileges(t *testing.T) {
 
 	go taskEngine.AddTask(testTask)
 
+	cancel := discardEvents(taskEvents)
```

Comment:
This test failed intermittently due to the logic here, this should be fixed soon as the single channel of events has been merged, I will change this to use the single channel.